test/compile/unit.test.ts

```diff
@@ -19,6 +19,18 @@ describe('UnitModel', function() {
         assert.equal(localLogger.warns[0], log.message.incompatibleChannel(SHAPE, BAR));
       }));
 
+    it('should drop invalid channel and throws warning', log.wrap((localLogger) => {
+        const INVALID_CHANNEL = '_y';
+        const model = parseUnitModel({
+          mark: 'bar',
+          encoding: {
+            _y: {type: 'quantitative'}
+          }
+        } as any); // To make parseUnitModel accept the model with invalid encoding channel
+        assert.equal(model.encoding.shape, undefined);
```

Comment:
I'm not sure why this is related to `model.encoding.shape`.

Reply:
Removed the assert.